j5/backends/hardware/sr/v4/serial/power_board.py

```diff
@@ -0,0 +1,354 @@
+
+        response = self.request(command)
+        if response is not None:
+            return response
+        else:
+            raise CommunicationError("Power board responded with ACK but expected data.")
+
+    # TODO: Can we cache this?
+    def get_identity(self) -> BoardIdentity:
+        """
+        Get the board identity information.
+
+        :raises CommunicationError: the identity response did not match the expected format.
+        :returns: A tuple of board identity information
+        """
+        response = self.request("*IDN?")  # TODO: Handle None here?
+        parts = response.split(":", 4)
+        if len(parts) != 4:
+            raise CommunicationError(f"Identify response did not match format: {response}")
+        else:
+            return BoardIdentity(*parts)
+
+    def reset_board(self) -> None:
+        self.request("*RESET")
+
+    @property
+    def firmware_version(self) -> str:
+        """
+        The firmware version reported by the board.
+
+        :returns: firmware version reported by the board, if any.
+        """
+        identity = self.get_identity()
+        return identity.software_version
+
+    def get_power_output_enabled(self, identifier: int) -> bool:
+        """
+        Get whether a power output is enabled.
+
+        :param identifier: power output to fetch status of.
+        :returns: status of the power output.
+        :raises ValueError: Invalid power output identifier.
+        """
+        if identifier in range(6):
+            response = self.request_with_response(f"OUT:{identifier}:GET?")
+            if response == "0":
+                return False
+            elif response == "1":
+                return True
+            else:
+                raise CommunicationError(f"Invalid response received: {response}")
+        else:
+            raise ValueError(f"Invalid identifier: {identifier!r}")
+
+    def set_power_output_enabled(
+        self, identifier: int, enabled: bool,
+    ) -> None:
+        """
+        Set whether a power output is enabled.
+
+        :param identifier: power output to enable / disable
+        :param enabled: status of the power output.
+        :raises ValueError: Invalid power output identifier.
+        """
+        if identifier in range(6):
+            state = "1" if enabled else "0"
+            self.request(f"OUT:{identifier}:SET:{state}")
+        else:
+            raise ValueError(f"Invalid identifier: {identifier!r}")
+
+    def get_power_output_current(self, identifier: int) -> float:
+        """
+        Get the current being drawn on a power output, in amperes.
+
+        :param identifier: power output to fetch current of.
+        :returns: current of the output.
+        :raises ValueError: Invalid power output identifier.
+        """
+        if identifier in range(6):
+            response = self.request_with_response(f"OUT:{identifier}:I?")
+            return float(response)
```

Comment:
This should divide the response by 1000. The serial returns the value in mA and this function says it returns Amps

Reply:
[c3989df](https://github.com/j5api/j5/pull/812/commits/c3989df945938fdcedc7e5ba6ff411edbaeb3df3)